core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -106,3 +109,196 @@ func testHeaderRLPBackwardsCompatibility(t *testing.T) {
 		assert.Equal(t, hdr, got)
 	})
 }
+
+func TestBodyRLPBackwardsCompatibility(t *testing.T) {
+	rng := ethtest.NewPseudoRand(0)
+
+	randTx := func() *Transaction { return NewTx(&LegacyTx{Nonce: rng.Uint64()}) }
+	randHdr := func() *Header { return &Header{ParentHash: rng.Hash()} }
+	randWithdraw := func() *Withdrawal { return &Withdrawal{Index: rng.Uint64()} }
+
+	// We build up test-case [Body] instances from the power set of each of
+	// these components.
+	txs := [][]*Transaction{
+		nil, {}, // Must be equivalent for non-optional field
+		{randTx()},
+		{randTx(), randTx()}, // Demonstrates nested lists
+	}
+	uncles := [][]*Header{
+		nil, {},
+		{randHdr()},
+		{randHdr(), randHdr()},
+	}
+	withdrawals := [][]*Withdrawal{
+		nil, {}, // Must be different for optional field
+		{randWithdraw()},
+		{randWithdraw(), randWithdraw()},
+	}
+
+	var bodies []*Body
+	for _, t := range txs {
+		for _, u := range uncles {
+			for _, w := range withdrawals {
+				bodies = append(bodies, &Body{t, u, w})
```

Comment:
use `tx` instead of `t`, it gets confusing with the `t *testing.T`

```suggestion
	for _, tx := range txs {
		for _, u := range uncles {
			for _, w := range withdrawals {
				bodies = append(bodies, &Body{tx, u, w})
```

Reply:
But then if I read it with a German accent it no longer sounds like I'm reciting the alphabet :(